soh/soh/OTRGlobals.cpp

```diff
@@ -2043,6 +2043,9 @@ extern "C" int CustomMessage_RetrieveIfExists(PlayState* play) {
         messageEntry = CustomMessageManager::Instance->RetrieveMessage(customMessageTableID, TEXT_HEART_PIECE);
         CustomMessageManager::ReplaceStringInMessage(messageEntry, "{{heartPieceCount}}", std::to_string(gSaveContext.sohStats.heartPieces + 1));
     }
+    if (textId == TEXT_MARKET_GUARD_NIGHT && CVar_GetS32("gMarketSneak", 0) && play->sceneNum != SCENE_MARKET_NIGHT) {
```

Comment:
> `play->sceneNum != SCENE_MARKET_NIGHT`

why is this checking that the scene *isn't* `SCENE_MARKET_NIGHT`

Reply:
There isn't any distinction between the guard near the gate and the guards found in the market otherwise (and it wouldn't make sense for the guards inside the market to let you out to the field, at least to me.)
The logic here is lifted from original programming, but this could be replaced with whether `sceneNum` is whatever enum market entrance is.

Reply:
> but this could be replaced with whether `sceneNum` is whatever enum market entrance is.

I'd say lets do that, I think it's `SCENE_ENTRA_N` but you should double check